libs/gson-support/src/main/java/org/codice/gsonsupport/GsonFactory.java

```diff
@@ -0,0 +1,204 @@
+    static final TypeAdapterFactory FACTORY =
+        new TypeAdapterFactory() {
+
+          public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
+            Type type = typeToken.getType();
+            Class<? super T> rawType = typeToken.getRawType();
+
+            if (Number.class.isAssignableFrom(rawType)) {
+              return (TypeAdapter<T>) new CustomTypeAdapter(gson);
+            }
+            if (Map.class.isAssignableFrom(rawType)) {
+              Type[] mapKeyAndValueTypes = GsonTypes.getMapKeyAndValueTypes(type, rawType);
+              return buildTypeAdapter(gson, mapKeyAndValueTypes[1]);
+            }
+            if (List.class.isAssignableFrom(rawType)) {
+              Type collectionType = GsonTypes.getCollectionElementType(type, rawType);
+              return buildTypeAdapter(gson, collectionType);
+            }
+
+            return null;
+          }
+
+          private <T> TypeAdapter<T> buildTypeAdapter(Gson gson, Type collectionType) {
+            if (TypeToken.get(collectionType).getRawType() == Object.class) {
+              return (TypeAdapter<T>) new CustomTypeAdapter(gson);
+            } else {
+              TypeAdapter<?> collectionTypeAdapter = gson.getAdapter(TypeToken.get(collectionType));
+              return (TypeAdapter<T>) new CustomTypeAdapter(gson, collectionTypeAdapter);
+            }
+          }
+        };
+
+    private final Gson gson;
+    private final TypeAdapter<?> collectionTypeAdapter;
+
+    public CustomTypeAdapter(Gson gson) {
+      this(gson, null);
+    }
+
+    public CustomTypeAdapter(Gson gson, TypeAdapter<?> collectionTypeAdapter) {
+      this.gson = gson;
+      this.collectionTypeAdapter = collectionTypeAdapter;
+    }
+
+    @Override
+    public Object read(JsonReader in) throws IOException {
+      final JsonToken token = in.peek();
+
+      switch (token) {
+        case BEGIN_ARRAY:
+          final List<Object> list = new ArrayList<>();
+
+          in.beginArray();
+          while (in.hasNext()) {
+            if (collectionTypeAdapter == null) {
+              list.add(read(in));
+            } else {
+              list.add(collectionTypeAdapter.read(in));
+            }
+          }
+          in.endArray();
+          return list;
+        case BEGIN_OBJECT:
+          final Map<String, Object> map = new LinkedMap<>();
+
+          in.beginObject();
+          while (in.hasNext()) {
+            if (collectionTypeAdapter == null) {
+              map.put(in.nextName(), read(in));
+            } else {
+              map.put(in.nextName(), collectionTypeAdapter.read(in));
+            }
+          }
+          in.endObject();
+          return map;
+
+        case STRING:
+          return in.nextString();
+        case NUMBER:
+          final String n = in.nextString();
+
+          if (n.indexOf('.') == -1) {
+            try {
+              if (n.endsWith("l") || n.endsWith("L")) {
```

Comment:
We should be able to parse as long in all cases. I'll test and verify.

As for the values ending in 'l' or 'L', these were discovered during the testing process - I've lost track of where - and may be moot at this point.

Reply:
I am ok keeping the support for "l" and "L". My concern was over adding Integer and no longer doing it as Long in all non floating point cases which I know would break migration.